### ISSUE TITLE
CA1810 should not warn when assigning field in event handler.

### ISSUE THREAD

**Issue by JohanLarsson** [dotnet/roslyn-analyzers#3138]
```cs
class C
{
    private static string? s;

    //     ↓ CA1810 should not warn here
    static C()
    {
        Console.CancelKeyPress += (o, e) => s = string.Empty;
    }
}
```

**Comment by Evangelink** [comment]
Hey @JohanLarsson,
Is your problem with the location of the issue or with the fact that an issue is reported for an event subscription case?

**Comment by JohanLarsson** [comment]
@Evangelink CA1810 should not warn suggesting removing the static ctor there. Would be different if the ctor contained a field assignment that could be moved to field initializer.

**Comment by Evangelink** [comment]
Makes sense to me as I don't think there is any other good way to handle this case. @mavasani is it ok for you?

**Comment by mavasani** [comment]
Yes, sounds reasonable to me.

**Comment by Evangelink** [comment]
@JohanLarsson out of curiosity, are you unsubscribing from the event somewhere?

**Comment by JohanLarsson** [comment]
@Evangelink the sample code is just a minimal sample using familiar types.

Also for your question unsubscribing is probably not critical there.

**Comment by Evangelink** [comment]
It's not :)

I was doing some experiments and if you create a method (static in this case) to set the field then you don't have the issue. Which made me realize that this is usually what I do for events to be able to do the unsubscription.
